### PR TITLE
Upgrade SQL-parser version

### PR DESCRIPTION
**Pull request by mweisgut**
This PR upgrades the SQL parser to the latest version, which requires some changes in the SQL translator.

### CODE REVIEW

**Comment by mweisgut** [comment]
The SQL parser contains a bug causing the CI crash. Patch: https://github.com/hyrise/sql-parser/pull/201

File: src/lib/sql/sql_translator.cpp [hunk 1]
```diff
@@ -1309,27 +1311,53 @@ std::shared_ptr<AbstractLQPNode> SQLTranslator::_translate_create_table(const hs
...
+          column_definition.data_type = DataType::Float;
+          break;
         case hsql::DataType::FLOAT:
           column_definition.data_type = DataType::Float;
           break;
         case hsql::DataType::DOUBLE:
           column_definition.data_type = DataType::Double;
           break;
         case hsql::DataType::CHAR:
+          std::cout << "WARNING: Ignoring the length of CHAR as Hyrise has no way of working with it\n";
```
Comment (Bouncner) [inline src/lib/sql/sql_translator.cpp:1338]: "Ignoring the length of CHAR. Hyrise's strings are not length-limited." ?

Comment (Bouncner) [reply src/lib/sql/sql_translator.cpp:1338]: I don't think the "no way of working with it" is not clear enough as a user information. Sounds like we don't know what to do with the text.

**Review by Bouncner — COMMENTED** [review]
Just two small issues (see review).

And is it "not yet supported" (CREATE INDEX) or is it "not supported yet" (date types)?

Besides, are there any tests we need with the latest SQL parser changes? I don't have an overview what parser changes get included with this update.

File: src/lib/sql/sql_translator.cpp [hunk 2]
```diff
@@ -1335,27 +1335,27 @@ std::shared_ptr<AbstractLQPNode> SQLTranslator::_translate_create_table(const hs
...
+          std::cout << "WARNING: Ignoring the length of CHAR. Hyrise's strings are not length-limited\n";
           column_definition.data_type = DataType::String;
           break;
         case hsql::DataType::VARCHAR:
-          std::cout << "WARNING: Ignoring the length of VARCHAR as Hyrise has no way of working with it\n";
+          std::cout << "WARNING: Ignoring the length of VARCHAR. Hyrise's strings are not length-limited\n";
           column_definition.data_type = DataType::String;
           break;
         case hsql::DataType::TEXT:
-          std::cout << "WARNING: Ignoring the length of TEXT as Hyrise has no way of working with it\n";
```
Comment (Bouncner) [inline src/lib/sql/sql_translator.cpp]: Wait, as far as I remember, TEXT does not have an optional size. At least in the major DBMSs. So actually, TEXT is pretty much what we do in Hyrise as the default.

**Comment by mweisgut** [comment]
> Besides, are there any tests we need with the latest SQL parser changes? I don't have an overview what parser changes get included with this update.

Additional features and changes of the SQL parser should be tested in its test suite accordingly.

For Hyrise, besides running the test suite successfully, I think we should check if the benchmarks (TPC-H, TPC-DS, JOB, JCC-H, and TPC-C) can be executed successfully. The TPC-H, TPC-DS, and JOB is checked in the CI already.

**Review by Bouncner — APPROVED** [review]

**Comment by mweisgut** [comment]
Check if the following benchmarks still run successfully with this PR
- [x] TPC-H
- [x] TPC-DS
- [x] TPC-C
- [x] JOB
- [ ] JCC-H (not checked yet)

**Comment by Bensk1** [comment]
> For Hyrise, besides running the test suite successfully, I think we should check if the benchmarks (TPC-H, TPC-DS, JOB, JCC-H, and TPC-C) can be executed successfully. The TPC-H, TPC-DS, and JOB are checked in the CI already.

FullCI should execute the TPCC (clang-release stage) and JCCH (system-tests stage) with verification.

**Pull request merged**